repos/system_upgrade/common/actors/systemfacts/libraries/systemfacts.py

```diff
@@ -217,7 +217,21 @@ def get_sysctls_status():
 
 def get_repositories_status():
     """ Get a basic information about YUM repositories installed in the system """
-    return RepositoriesFacts(repositories=repofileutils.get_parsed_repofiles())
+    try:
+        return RepositoriesFacts(repositories=repofileutils.get_parsed_repofiles())
+    except repofileutils.InvalidRepoDefinition as e:
+        raise StopActorExecutionError(
+            message='Invalid repository definition',
+            details={
+                'details': str(e),
+                'file': e.repofile,
+                'repoid': e.repoid,
+                'hint': 'To resolve this issue, please visit: {url}'
+                        .format(
+                            url='https://access.redhat.com/solutions/3185891'
+                        )
+            }
```

Comment:
There is a problem with this approach, if the `InvalidRepoDefinition` exception is raised and not caught and handled the way it's handled here, there will be no message (and possibly no other info besides traceback, not sure how the kwargs would be displayed).

You could move the message definition into the InvalidRepoDefinition constructor, or at least to the ctor call site. Also the message could contain the repoid and repofile directly in the string too, e.g. `Invalid repository definition: <repoid> in: <filepath>`. But it's good having them available as members still.

Personally I like the way it's done here: https://github.com/oamg/leapp-repository/blob/236483d5478ce845726ee05b7ca3e080105cdb80/repos/system_upgrade/el7toel8/libraries/isccfg.py#L11-L23
but keeping that up to you.

Reply:
It makes more sense! Put it there, and again tested it.